src/movement/FMIMovement.java

```diff
@@ -74,7 +69,10 @@ public FMIMovement(FMIMovement other) {
         nodeGrid = other.nodeGrid;
         pointsOfInterest = other.pointsOfInterest;
         pathFinder = other.pathFinder;
-        schedule = new Schedule(other.eventsByTimeslot, rng);
+        ArrayList<MapNode> collectionAreas = pointsOfInterest.stream()
+                .filter(location -> location.isType(NodeType.COLLECTION_AREA.getType()))
+                .collect(Collectors.toCollection(ArrayList::new));
```

Comment:
This could be computed once in the prototype constructor

Reply:
True, on the other hand we would need a variable then, of which we already have defined a superset (pointsOfInterest), which felt functionally redundant.